portfolio/src/main/java/com/google/sps/servlets/NewCommentServlet.java

```diff
@@ -27,27 +36,70 @@
 
   /**
    * {@inheritDoc}
-   * 
-   * <p>This Method handles POST requests corresponding to a new comment and creates a new Entity 
+   *
+   * <p>This Method handles POST requests corresponding to a new comment and creates a new Entity
    * for that comment in the Google Cloud Datastore.
    *
-   * <p>The POST request also results in a re-direct back to the original server-dev page.
-   * TODO(Issue #15): Do verfification on a new comment before adding it to the comments list.
+   * <p>This POST request originates from the 'new comment' form in server-dev.html and is initially
+   * sent to Blobstore for file processing. Once the Blobstore forward the request to this servlet,
+   * the name of file submitted in the form can be used to get the corresponding blob key to be 
+   * stored in the Datastore
+   * Blobstore. The POST request also results in a re-direct back to the original server-dev page.
+   *
+   * <p>TODO(Issue #15): Do verfification on a new comment before adding it to the comments list.
    */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String newComment = request.getParameter("comment");
     long timestamp = System.currentTimeMillis();
+    BlobKey blobKey = getBlobKey(request, "image");
 
     Entity taskEntity = new Entity("Comment");
     taskEntity.setProperty("text", newComment);
     taskEntity.setProperty("timestamp", timestamp);
-
+    taskEntity.setProperty("blobKey", blobKey);
     datastore.put(taskEntity);
 
     response.sendRedirect("/pages/server-dev.html");
   }
+
+  /**
+   * Returns a BlobKey object corresponding to the uploaded file.
+   *
+   * @param request The <code>HttpServletRequest</code> for the POST request.
+   * @param formInputElementName The name attribute of the image file input to the form.
+   * @return The blob key associated with the uploaded image file. Null is returned if 
+   *         the user did not select a file or the file is not an image type.
+   */
+  private BlobKey getBlobKey(HttpServletRequest request, String formInputElementName) {
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get(formInputElementName);
+
+    // User submitted form without selecting a file, so we can't get a URL. (dev server)
+    if (blobKeys == null || blobKeys.isEmpty()) {
+      return null;
+    }
+
+    // Our form only contains a single file input, so get the first index.
+    BlobKey blobKey = blobKeys.get(0);
+
+    // User submitted form without selecting a file, so we can't get a URL. (live server)
+    BlobInfo blobInfo = new BlobInfoFactory().loadBlobInfo(blobKey);
+    if (blobInfo.getSize() == 0) {
+      blobstoreService.delete(blobKey);
+      return null;
+    }
+
+    // Check the validity of the file here by making sure it's an image file
+    if (!"image".equals(blobInfo.getContentType().substring(0, 5))) {
```

Comment:
Should mention this in previous PRs. You can do something like this 
```suggestion
    if (blobInfo.getContentType().startsWith("image"))) {
```

Reply:
Ohh I did not realize this existed. I like it though because it reminds me of python. Done.